### PR TITLE
Allow whitespace and comments in db key file

### LINKED ISSUE #4542: Allow whitespace and comments in db key file
It would be useful if the db key file parsed generously around additional whitespace.

Also in order to help guide an admin while rotating keys, we should allow to add comments.

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #4542

### CODE REVIEW

**Comment by dralley** [comment]
I understand allowing whitespace, but why comments?  This should probably have an issue with additional explanation

**Comment by mdellweg** [comment]
> I understand allowing whitespace, but why comments? This should probably have an issue with additional explanation

Whitespace was triggered by this: https://discourse.pulpproject.org/t/pulp-minimal-3-35-pulp-api-metadata-signing-service-http-1-1-500-internal-server-error/1002/5

Comments were like "why not". And my thinking is: You can specify multiple keys in there (to allow key rotation.) Adding comments which key is supposed to be the new one and must move to the top of the list eventually or even a note when it's supposed to happen may become priceless.

File: CHANGES/+db_encryption.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Parsing of the db encryption key was made resiliant to whitespace and to allow comments.
```
Comment (ipanova) [inline CHANGES/+db_encryption.feature:1]: can you file an issue so the changelog is in the format of XXXX.feature?

**Review by ipanova — APPROVED** [review]

File: CHANGES/+db_encryption.feature (hunk 1 above)
Comment (mdellweg) [reply CHANGES/+db_encryption.feature:1]: Sure. I wanted to know if we would accept this as a feature...

Comment (ipanova) [reply CHANGES/+db_encryption.feature]: Any step towards making things more resilient is a good step, I think.

**Review by ggainey — APPROVED** [review]

**Pull request merged**